tests/pytests/test_vecsim.py

```diff
@@ -930,4 +932,79 @@ def test_fail_on_v1_dialect():
+
+    # Change the text value to 'other' for 20% of the vectors (with ids 5, 10, ..., index_size)
+    for i in range(1, int(index_size/5) + 1):
+        vector = np.full(dim, 5*i, dtype='float32')
+        conn.execute_command('HSET', 5*i, 'v', vector.tobytes(), 't', 'other')
+
+    # Run VecSim query in hybrid mode, expect to get only the vectors that passes the filters
+    # (index_size-10 and index_size-100, since they has "other" in its 't' field and its id is in inkeys list).
+    inkeys = [index_size-2, index_size-10, index_size-100]
+    expected_res = [2, str(index_size-100), ['__v_score', str(dim*100**2)], str(index_size-10), ['__v_score', str(dim*10**2)]]
+    env.expect('FT.SEARCH', 'idx', '(other)=>[KNN 10 @v $vec_param]', 'INKEYS', len(inkeys), *inkeys,
+               'RETURN', 1, '__v_score', 'PARAMS', 2, 'vec_param', query_data.tobytes()).equal(expected_res)
+
+    # Test legacy numeric and geo global filters
+    expected_res = [10]
+    # Expect to get top 10 ids where maximum is index_size/2 in the index (due to the numeric filter).
+    for i in range(10):
+        expected_res.append(str(int(index_size/2-i)))
+        expected_res.append(['__v_score', str(int(dim*(index_size/2+i)**2))])
+    env.expect('FT.SEARCH', 'idx', '*=>[KNN 10 @v $vec_param]', 'filter', 'num', 0, index_size/200, 'SORTBY', '__v_score',
+               'RETURN', 1, '__v_score', 'PARAMS', 2, 'vec_param', query_data.tobytes()).equal(expected_res)
+
+    # Expect to get top 10 ids where maximum is 31 in the index (due to the geo filter).
+    expected_res = [10]
+    for i in range(10):
+        expected_res.append(str(31-i))
+        expected_res.append(['coordinate', str((31-i)/100)+","+str((31-i)/100)])
+    env.expect('FT.SEARCH', 'idx', '*=>[KNN 10 @v $vec_param]', 'geofilter', 'coordinate', 0.0, 0.0, 50, 'km',
+               'SORTBY', '__v_score', 'RETURN', 1, 'coordinate',
+               'PARAMS', 2, 'vec_param', query_data.tobytes()).equal(expected_res)
+
+    # Test complex query with multiple global filters - this query applies 4 global filters:
+    # 2 numeric filters - the second filter is a subset of the first one - enforces that the result ids
+    # will be between index_size/3 to index_size/2.
+    # geo filter - get all ids whose coordinates are in 50 km radius from (5.0, 5.0) - will enforce that the result ids
+    # are in the range of (index_size/2 - 31, index_size/2 + 31), according to the previous query and the fact that the
+    # coordinates of index_size/2 are (5.0, 5.0).
+    # Finally, inkeys filter enforces that only ids that multiply by 7 are valid.
+    # On top of all that, we have the hybrid query that filters out ids that multiply by 5
+    # (i.e., their text field is not 'hybrid') - and this is the reason for the expected ids
+    expected_res = ([str(i) for i in range(int(index_size/2), int(index_size/2) - 32, -1)
+                    if i % 5 != 0 and i % 7 == 0])
+    expected_res.insert(0, len(expected_res))
```

Comment:
`index_size/200` and 300 compared to the comment you wrote 2 and 3.
Also, do you get 5 results?

Reply:
no, there only 4 results
about the num values - changed following previous comment